File: soal5.ts
function angryProfessor(k: number, a: number[]): string {
    // Hitung jumlah siswa yang datang tepat waktu atau lebih awal (a[i] <= 0)
    const onTimeStudents = a.filter(arrival => arrival <= 0).length;

    // Jika jumlah siswa yang datang tepat waktu >= k, kelas tidak dibatalkan
    return onTimeStudents >= k ? "NO" : "YES";
}

// Contoh penggunaan
const testCases: number = parseInt(prompt("Masukkan jumlah test case: ") || "0");

for (let i = 0; i < testCases; i++) {
    const [n, k] = prompt("Masukkan jumlah siswa dan batas minimal hadir: ")?.split(" ").map(Number) || [0, 0];
    const a: number[] = prompt("Masukkan daftar kedatangan siswa: ")?.split(" ").map(Number) || [];

    console.log(angryProfessor(k, a));
}

File: soal5.py
def angryProfessor(k, a):
    # Hitung jumlah siswa yang datang tepat waktu atau lebih awal (a[i] <= 0)
    on_time_students = sum(1 for arrival in a if arrival <= 0)

    # Jika jumlah siswa yang datang tepat waktu >= k, kelas tidak dibatalkan
    return "NO" if on_time_students >= k else "YES"

# Contoh penggunaan
if __name__ == "__main__":
    t = int(input("Masukkan jumlah test case: ").strip())

    for _ in range(t):
        n, k = map(int, input().split())  # Baca n (jumlah siswa) dan k (batas minimal siswa hadir)
        a = list(map(int, input().split()))  # Baca daftar kedatangan siswa

        print(angryProfessor(k, a))
